var/spack/repos/builtin/packages/neurodamus-core/package.py

```diff
@@ -42,17 +46,44 @@ def install(self, spec, prefix):
         shutil.copytree('mod', prefix.mod)
         if spec.satisfies('+python'):
             copy_tree('resources/neurodamus-py', prefix.python)
-        if spec.satisfies('+common'):
-            copy_all('resources/common/hoc', prefix.hoc)
-            copy_all('resources/common/mod', prefix.mod)
 
         filter_file(r'UNKNOWN_CORE_VERSION', r'%s' % spec.version, prefix.hoc.join('defvar.hoc'))
-
         try:
             commit_hash = self.fetcher[0].get_commit()
             filter_file(r'UNKNOWN_CORE_HASH', r'\'%s\'' % commit_hash, prefix.hoc.join('defvar.hoc'))
         except Exception as e:
-            tty.warn(e)
+            tty.warn(str(e))
+
+        # +Common will bring common mods and build a bare nrnmechlib
+        if spec.satisfies('+common'):
+            copy_all('resources/common/hoc', prefix.hoc)
+            copy_all('resources/common/mod', prefix.mod)
+
+            with working_dir(prefix):
+                which('nrnivmodl')('-incflags', '-DDISABLE_REPORTINGLIB', 'mod')
+
+                bindir = os.path.basename(self.neuron_archdir)
+                special = join_path(bindir, 'special')
+                open('.bindir', 'w').write(bindir)
+
+                if not os.path.isfile(special):
+                    raise Exception("Failed to build neurodamus core mods")
```

Comment:
See [sanity_check_is*](https://spack.readthedocs.io/en/latest/packaging_guide.html#sanity-check-is-file-and-sanity-check-is-dir)

Reply:
Fancy! But can we do it conditionally? `when='+common'` ?

Reply:
I don't know that :) if it's not documented, we can use it like this.

Reply:
Apparently not. Implementation is still pretty simple.
https://github.com/spack/spack/search?q=sanity_check_is_file&unscoped_q=sanity_check_is_file